routing/router.go

```diff
@@ -1203,14 +1203,16 @@ func (r *ChannelRouter) processUpdate(msg interface{}) error {
 		// Now that we know this isn't a stale update, we'll apply the
 		// new edge policy to the proper directional edge within the
 		// channel graph.
-		if err = r.cfg.Graph.UpdateEdgePolicy(msg); err != nil {
-			err := errors.Errorf("unable to add channel: %v", err)
-			log.Error(err)
-			return err
-		}
+		if exists {
```

Comment:
Is this a fix in itself? Ideally this could be yet another separate commit, but if you find it too small for that, it's fine with me.

Reply:
I _think_ maybe the proper fix here is something like this: https://github.com/lightningnetwork/lnd/pull/2279/commits/ad92e5c9638c95b53f8756d462424d6317de76bf (feel free to cherry-pick it into this PR if needed).

I believe then you wouldn't have to flip the `AssumeChannelValid` in the config during tests?

Reply:
This can be removed now that #2549 is merged.